Source/Core/Common/SymbolDB.h

```diff
@@ -87,27 +88,49 @@ class SymbolDB
   SymbolDB();
   virtual ~SymbolDB();
 
-  virtual Symbol* GetSymbolFromAddr(u32 addr) { return nullptr; }
-  virtual Symbol* AddFunction(const Core::CPUThreadGuard& guard, u32 start_addr) { return nullptr; }
+  virtual const Symbol* GetSymbolFromAddr(u32 addr) const { return nullptr; }
+  virtual const Symbol* AddFunction(const Core::CPUThreadGuard& guard, u32 start_addr)
+  {
+    return nullptr;
+  }
   void AddCompleteSymbol(const Symbol& symbol);
+  bool RenameSymbol(const Symbol& symbol, const std::string& symbol_name);
+  bool RenameSymbol(const Symbol& symbol, const std::string& symbol_name,
+                    const std::string& object_name);
+
+  const Symbol* GetSymbolFromName(std::string_view name) const;
+  std::vector<const Symbol*> GetSymbolsFromName(std::string_view name) const;
+  const Symbol* GetSymbolFromHash(u32 hash) const;
+  std::vector<const Symbol*> GetSymbolsFromHash(u32 hash) const;
+
+  template <typename F>
+  void ForEachSymbol(F f) const
+  {
+    std::lock_guard lock(m_mutex);
+    for (auto [addr, symbol] : m_functions)
```

Comment:
```suggestion
    for (const auto& [addr, symbol] : m_functions)
```
To avoid copying all those strings.